### PR TITLE
T1435 Remove useURL option from image2 plugin

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/1435-remove-isimage-option-from-image2-plugin

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/55bbe7af554e228/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/55bbe7af554e228/output.txt

Total script time: 0.59 mins

Server is running on http://pr444.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_cleanup** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/47108b2f6518c60/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/47108b2f6518c60/output.txt

Total script time: 0.47 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/79b28fc90dc4784/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/79b28fc90dc4784/output.txt

Total script time: 0.25 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/64a0fc0675aac97/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/64a0fc0675aac97/output.txt

Total script time: 0.84 mins

Server is running on http://pr444.butyr.org/templates/test.html

File: templates/supported/shared/editors/photo-editor.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 (function(){
   var _comm = new Comm(),
-      _manifest = {};
+      _manifest = {},
+      dataURI;
```
Comment (cadecairos) [inline templates/supported/shared/editors/photo-editor.js]: This should be initialized to `""`

File: templates/supported/shared/editors/photo-editor.js [hunk 2]
```diff
@@ -82,20 +87,21 @@
...
-        element.value = e.data[ item ];
-        if ( item === "src" ) {
-          if ( e.data[ "useURL" ] === false ) {
-            document.getElementById("src-container").style.display = "none";
-            //document.getElementById("drop-target").style.display = "block";
-          } else if ( e.data[ "useURL" ] === true ) {
-            document.getElementById("src-container").style.display = "block";
-            //document.getElementById("drop-target").style.display = "none";
-          }
+        var element = document.getElementById( item ),
```
Comment (cadecairos) [inline templates/supported/shared/editors/photo-editor.js]: the inside of the above for..in should get wrapped in an `if( _manifest.hasOwnProperty( item ) ) {}` block for good measure

File: templates/supported/shared/editors/photo-editor.js [hunk 3]
```diff
@@ -160,8 +166,8 @@
                 elem.appendChild( option );
               }
               elem.value = this.defaultValue( manifestItem, popcornOptions[ manifestProp ] );
-              if ( manifestProp === "target" ) { 
-                elem.value = masterTarget 
+              if ( manifestProp === "target" ) {
+                elem.value = masterTarget
```
Comment (cadecairos) [inline templates/supported/shared/editors/photo-editor.js]: missing a semicolon here

File: templates/supported/shared/editors/photo-editor.js [hunk 4]
```diff
@@ -217,20 +222,18 @@
...
-        canvas.id = "grabimage";
-
         dropTarget = document.createElement( "div" );
         dropTarget.id = "drop-target";
         dropTarget.innerHTML = "<span>Drag an image from your desktop...</span>";
 
-        if(  popcornOptions[ item ] ) { 
-          dropTarget.style.backgroundImage = "url('" + popcornOptions[ item ] + "')"; 
-        }  
+        if(  popcornOptions[ item ] && popcornOptions[ item ].substring( 0, 10 ) === "data:image" ) {
```
Comment (cadecairos) [inline templates/supported/shared/editors/photo-editor.js]: use `/^data:image/.test( popcornOptions[ item ] )` to check for the string

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_cleanup** from @cadecairos received. Current queue size: 2

Live output at: http://botio.butyr.org:8000/40098ca1f77ce41/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/40098ca1f77ce41/output.txt

Total script time: 0.21 mins

File: templates/supported/shared/editors/photo-editor.js (hunk 4 above)
Comment (dseif) [inline templates/supported/shared/editors/photo-editor.js]: or popcornOptions[ item ].indexOf( "data:Image" )

File: templates/supported/newscaster/butter/template.js [hunk 5]
```diff
@@ -69,21 +69,17 @@ document.addEventListener( "DOMContentLoaded", function( e ){
...
+                  stop: function( event, ui ) {
                     _container.style.border = "";
-                    trackEvent.update({ height: ui.size.height + "px", width: ui.size.width + "px" });
+                    trackEvent.update( { height: ui.size.height + "px", width: ui.size.width + "px" } );
                   }
                 }).draggable({
-                  stop: function(event, ui) {
-                    trackEvent.update({top: ui.position.top + "px", left: ui.position.left + "px" });
+                  stop: function( event, ui ) {
+                    trackEvent.update( {top: ui.position.top + "px", left: ui.position.left + "px" } );
```
Comment (dseif) [inline templates/supported/newscaster/butter/template.js]: you actually made the style of this line wrong ( it should actually be multilined if you're trying to make it correct )

File: templates/supported/shared/editors/photo-editor.js [hunk 6]
```diff
@@ -47,8 +49,12 @@
         if ( options && options[ item ] ) {
            popcornOptions[ item ] = options[ item ];
         } else if ( item === "src" ) {
-          if ( document.getElementById( "useURL" ) && document.getElementById( "useURL" ).value === true ) {
-            popcornOptions[ item ] = document.getElementById( item ).value;
+          var source = document.getElementById( item ).value;
+          if ( /^data:image/.test( source ) ) {
+            popcornOptions[ item ] = dataURI;
+          }
+          else {
```
Comment (dseif) [inline templates/supported/shared/editors/photo-editor.js]: else should be on the line above `} else {`

File: templates/supported/shared/editors/photo-editor.js [hunk 7]
```diff
@@ -82,20 +88,35 @@
...
+              // Store the data URI so we can still keep it correct for the popcornOptions
+              dataURI = data;
+              // Only display data/image if the source is a dataURI as anything longer is confusing
+              // And chrome can't handle really long data URIs in textfields
+              element.value = data.substring( 0, 10 );
+              // Set the src of the drop box in the editor to the dataURI
+              dropTarget.innerHTML = "";
+              dropTarget.style.backgroundImage = "url('" + data + "')";
+            }
+            else {
```
Comment (dseif) [inline templates/supported/shared/editors/photo-editor.js]: same here

File: templates/supported/shared/editors/photo-editor.js [hunk 8]
```diff
@@ -82,20 +88,35 @@
...
+              element.value = data;
+
+              dropTarget.innerHTML = DROP_AREA_TEXT;
+              dropTarget.style.backgroundImage = "";
+            }
+          }
+          else {
```
Comment (dseif) [inline templates/supported/shared/editors/photo-editor.js]: and here

**Pull request merged**